### PR TITLE
Test against Rails v7.1, v7.2, and v8.0

### PR DESCRIPTION
**Pull request by leoarnold**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,8 +30,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ruby: ["3.1", "3.2"]
-        rails: ["6.1", "7.0"]
+        ruby: ["3.2", "3.3"]
+        rails: ["6.1", "7.0", "7.1", "7.2", "8.0"]
```
Comment (calebhearth) [inline .github/workflows/ci.yml:34]: Looks like we can probably drop 6.1 and 7.0. I don't mind dropping 7.0 a few months early personally.

https://endoflife.date/rails

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
```
Comment (calebhearth) [inline .github/workflows/ci.yml:16]: I don't particularly mind this but is there a reason for the bump?

Comment (leoarnold) [reply .github/workflows/ci.yml:16]: Yes, see e.g. the warnings at the end of this page: https://github.com/scenic-views/scenic/actions/runs/9181920538

Comment (calebhearth) [reply .github/workflows/ci.yml:16]: Nice, thanks.

**Pull request merged**